<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html>
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>celda Cell and Gene Clustering Model — celda_CG • celda</title>

<!-- jquery -->
<script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://maxcdn.bootstrapcdn.com/bootswatch/3.3.7/yeti/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">

<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script>

<!-- sticky kit -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



<meta property="og:title" content="celda Cell and Gene Clustering Model — celda_CG" />

<meta property="og:description" content="celda Cell and Gene Clustering Model" />
<meta name="twitter:card" content="summary" />



<!-- mathjax -->
<script src='https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML'></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">celda</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/celda-analysis.html">Analyzing single-cell RNA-seq count data with Celda</a>
    </li>
  </ul>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>celda Cell and Gene Clustering Model</h1>
    
    <div class="hidden name"><code>celda_CG.Rd</code></div>
    </div>

    <div class="ref-description">
    
    <p>celda Cell and Gene Clustering Model</p>
    
    </div>

    <pre class="usage"><span class='fu'>celda_CG</span>(<span class='no'>counts</span>, <span class='kw'>sample.label</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='no'>K</span>, <span class='no'>L</span>, <span class='kw'>alpha</span> <span class='kw'>=</span> <span class='fl'>1</span>, <span class='kw'>beta</span> <span class='kw'>=</span> <span class='fl'>1</span>,
  <span class='kw'>delta</span> <span class='kw'>=</span> <span class='fl'>1</span>, <span class='kw'>gamma</span> <span class='kw'>=</span> <span class='fl'>1</span>, <span class='kw'>algorithm</span> <span class='kw'>=</span> <span class='fu'>c</span>(<span class='st'>"EM"</span>, <span class='st'>"Gibbs"</span>), <span class='kw'>stop.iter</span> <span class='kw'>=</span> <span class='fl'>10</span>,
  <span class='kw'>max.iter</span> <span class='kw'>=</span> <span class='fl'>200</span>, <span class='kw'>split.on.iter</span> <span class='kw'>=</span> <span class='fl'>10</span>, <span class='kw'>split.on.last</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
  <span class='kw'>seed</span> <span class='kw'>=</span> <span class='fl'>12345</span>, <span class='kw'>nchains</span> <span class='kw'>=</span> <span class='fl'>3</span>, <span class='kw'>initialize</span> <span class='kw'>=</span> <span class='fu'>c</span>(<span class='st'>"random"</span>, <span class='st'>"split"</span>),
  <span class='kw'>count.checksum</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>z.init</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>y.init</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>logfile</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>verbose</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>)</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>counts</th>
      <td><p>Integer matrix. Rows represent features and columns represent cells.</p></td>
    </tr>
    <tr>
      <th>sample.label</th>
      <td><p>Vector or factor. Denotes the sample label for each cell (column) in the count matrix.</p></td>
    </tr>
    <tr>
      <th>K</th>
      <td><p>Integer. Number of cell populations.</p></td>
    </tr>
    <tr>
      <th>L</th>
      <td><p>Integer. Number of feature modules.</p></td>
    </tr>
    <tr>
      <th>alpha</th>
      <td><p>Numeric. Concentration parameter for Theta. Adds a pseudocount to each cell population in each sample. Default 1.</p></td>
    </tr>
    <tr>
      <th>beta</th>
      <td><p>Numeric. Concentration parameter for Phi. Adds a pseudocount to each feature module in each cell population. Default 1.</p></td>
    </tr>
    <tr>
      <th>delta</th>
      <td><p>Numeric. Concentration parameter for Psi. Adds a pseudocount to each feature in each module. Default 1.</p></td>
    </tr>
    <tr>
      <th>gamma</th>
      <td><p>Numeric. Concentration parameter for Eta. Adds a pseudocount to the number of features in each module. Default 1.</p></td>
    </tr>
    <tr>
      <th>algorithm</th>
      <td><p>String. Algorithm to use for clustering cell subpopulations. One of 'EM' or 'Gibbs'. Default 'EM'.</p></td>
    </tr>
    <tr>
      <th>stop.iter</th>
      <td><p>Integer. Number of iterations without improvement in the log likelihood to stop inference. Default 10.</p></td>
    </tr>
    <tr>
      <th>max.iter</th>
      <td><p>Integer. Maximum number of iterations of Gibbs sampling to perform. Default 200.</p></td>
    </tr>
    <tr>
      <th>split.on.iter</th>
      <td><p>Integer. On every `split.on.iter` iteration, a heuristic will be applied to determine if a cell population or feature module should be reassigned and another cell population or feature module should be split into two clusters. To disable splitting, set to -1. Default 10.</p></td>
    </tr>
    <tr>
      <th>split.on.last</th>
      <td><p>Integer. After the the chain has converged, according to `stop.iter`, a heuristic will be applied to determine if a cell population or feature module should be reassigned and another cell population or feature module should be split into two clusters. If a split occurs, then 'stop.iter' will be reset. Default TRUE.</p></td>
    </tr>
    <tr>
      <th>seed</th>
      <td><p>Integer. Passed to set.seed(). Default 12345.</p></td>
    </tr>
    <tr>
      <th>nchains</th>
      <td><p>Integer. Number of random cluster initializations. Default 1.</p></td>
    </tr>
    <tr>
      <th>initialize</th>
      <td><p>Chararacter. One of 'random' or 'split'. With 'random', cells and features are randomly assigned to a clusters. With 'split' cell and feature clusters will be recurssively split into two clusters using `celda_C` and `celda_G`, respectively, until the specified K and L is reached. Default 'random'.</p></td>
    </tr>
    <tr>
      <th>count.checksum</th>
      <td><p>Character. An MD5 checksum for the `counts` matrix. Default NULL.</p></td>
    </tr>
    <tr>
      <th>z.init</th>
      <td><p>Integer vector. Sets initial starting values of z. If NULL, starting values for each cell will be randomly sampled from 1:K. 'z.init' can only be used when 'initialize' = "random". Default NULL.</p></td>
    </tr>
    <tr>
      <th>y.init</th>
      <td><p>Integer vector. Sets initial starting values of y. If NULL, starting values for each feature will be randomly sampled from 1:L. 'y.init' can only be used when 'initialize' = "random". Default NULL.</p></td>
    </tr>
    <tr>
      <th>logfile</th>
      <td><p>Character. Messages will be redirected to a file named `logfile`. If NULL, messages will be printed to stdout.  Default NULL.</p></td>
    </tr>
    <tr>
      <th>verbose</th>
      <td><p>Logical. Whether to print log messages. Default TRUE.</p></td>
    </tr>
    </table>
    
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>An object of class celda_CG with clustering results and various sampling statistics.</p>
    

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='no'>celda.sim</span> <span class='kw'>=</span> <span class='fu'><a href='simulateCells.html'>simulateCells</a></span>(<span class='kw'>model</span><span class='kw'>=</span><span class='st'>"celda_CG"</span>)
<span class='no'>celda.mod</span> <span class='kw'>=</span> <span class='fu'>celda_CG</span>(<span class='no'>celda.sim</span>$<span class='no'>counts</span>, <span class='kw'>K</span><span class='kw'>=</span><span class='no'>celda.sim</span>$<span class='no'>K</span>, <span class='kw'>L</span><span class='kw'>=</span><span class='no'>celda.sim</span>$<span class='no'>L</span>,
                     <span class='kw'>sample.label</span><span class='kw'>=</span><span class='no'>celda.sim</span>$<span class='no'>sample.label</span>, <span class='kw'>nchains</span><span class='kw'>=</span><span class='fl'>1</span>)</div><div class='output co'>#&gt; <span class='message'>--------------------------------------------------------------------</span></div><div class='output co'>#&gt; <span class='message'>Starting Celda_CG: Clustering cells and genes.</span></div><div class='output co'>#&gt; <span class='message'>--------------------------------------------------------------------</span></div><div class='output co'>#&gt; <span class='message'>Thu Sep 06 12:56:26 2018 .. Initializing chain 1 with 'random' (seed=12345)</span></div><div class='output co'>#&gt; <span class='message'>Thu Sep 06 12:56:26 2018 .... Completed iteration: 1 | logLik: -1302025.22571701</span></div><div class='output co'>#&gt; <span class='message'>Thu Sep 06 12:56:26 2018 .... Completed iteration: 2 | logLik: -1274189.9192103</span></div><div class='output co'>#&gt; <span class='message'>Thu Sep 06 12:56:26 2018 .... Completed iteration: 3 | logLik: -1259837.66674178</span></div><div class='output co'>#&gt; <span class='message'>Thu Sep 06 12:56:26 2018 .... Completed iteration: 4 | logLik: -1259840.6303583</span></div><div class='output co'>#&gt; <span class='message'>Thu Sep 06 12:56:26 2018 .... Completed iteration: 5 | logLik: -1259839.47116003</span></div><div class='output co'>#&gt; <span class='message'>Thu Sep 06 12:56:27 2018 .... Completed iteration: 6 | logLik: -1259838.95001855</span></div><div class='output co'>#&gt; <span class='message'>Thu Sep 06 12:56:27 2018 .... Completed iteration: 7 | logLik: -1259839.89588622</span></div><div class='output co'>#&gt; <span class='message'>Thu Sep 06 12:56:27 2018 .... Completed iteration: 8 | logLik: -1259841.14843277</span></div><div class='output co'>#&gt; <span class='message'>Thu Sep 06 12:56:27 2018 .... Completed iteration: 9 | logLik: -1259839.25990463</span></div><div class='output co'>#&gt; <span class='message'>Thu Sep 06 12:56:27 2018 .... Determining if any cell clusters should be split.</span></div><div class='output co'>#&gt; <span class='message'>Thu Sep 06 12:56:27 2018 .... Cluster 1 was reassigned and cluster 4 was split in two.</span></div><div class='output co'>#&gt; <span class='message'>Thu Sep 06 12:56:27 2018 .... Determining if any gene clusters should be split.</span></div><div class='output co'>#&gt; <span class='message'>Thu Sep 06 12:56:27 2018 .... Cluster 2 was reassigned and cluster 3 was split in two.</span></div><div class='output co'>#&gt; <span class='message'>Thu Sep 06 12:56:27 2018 .... Completed iteration: 10 | logLik: -1248291.50203677</span></div><div class='output co'>#&gt; <span class='message'>Thu Sep 06 12:56:28 2018 .... Completed iteration: 11 | logLik: -1248291.50203677</span></div><div class='output co'>#&gt; <span class='message'>Thu Sep 06 12:56:28 2018 .... Completed iteration: 12 | logLik: -1248291.50203677</span></div><div class='output co'>#&gt; <span class='message'>Thu Sep 06 12:56:28 2018 .... Completed iteration: 13 | logLik: -1248291.50203677</span></div><div class='output co'>#&gt; <span class='message'>Thu Sep 06 12:56:28 2018 .... Completed iteration: 14 | logLik: -1248291.50203677</span></div><div class='output co'>#&gt; <span class='message'>Thu Sep 06 12:56:28 2018 .... Completed iteration: 15 | logLik: -1248291.50203677</span></div><div class='output co'>#&gt; <span class='message'>Thu Sep 06 12:56:28 2018 .... Completed iteration: 16 | logLik: -1248291.50203677</span></div><div class='output co'>#&gt; <span class='message'>Thu Sep 06 12:56:28 2018 .... Completed iteration: 17 | logLik: -1248291.50203677</span></div><div class='output co'>#&gt; <span class='message'>Thu Sep 06 12:56:28 2018 .... Completed iteration: 18 | logLik: -1248291.50203677</span></div><div class='output co'>#&gt; <span class='message'>Thu Sep 06 12:56:28 2018 .... Completed iteration: 19 | logLik: -1248291.50203677</span></div><div class='output co'>#&gt; <span class='message'>Thu Sep 06 12:56:28 2018 .... Determining if any cell clusters should be split.</span></div><div class='output co'>#&gt; <span class='message'>Thu Sep 06 12:56:28 2018 .... No additional splitting was performed.</span></div><div class='output co'>#&gt; <span class='message'>Thu Sep 06 12:56:28 2018 .... Determining if any gene clusters should be split.</span></div><div class='output co'>#&gt; <span class='message'>Thu Sep 06 12:56:28 2018 .... No additional splitting was performed.</span></div><div class='output co'>#&gt; <span class='message'>Thu Sep 06 12:56:28 2018 .... Completed iteration: 20 | logLik: -1248291.50203677</span></div><div class='output co'>#&gt; <span class='message'>Thu Sep 06 12:56:28 2018 .. Finished chain 1 with seed 12345</span></div><div class='output co'>#&gt; <span class='message'>--------------------------------------------------------------------</span></div><div class='output co'>#&gt; <span class='message'>Completed Celda_CG. Total time: 1.817975 secs</span></div><div class='output co'>#&gt; <span class='message'>--------------------------------------------------------------------</span></div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      
      <li><a href="#value">Value</a></li>
      
      <li><a href="#examples">Examples</a></li>
    </ul>

  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Joshua Campbell, Sean Corbett, Yusuke Koga.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
   </div>

  

  </body>
</html>

