<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html>
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Function reference â€¢ celda</title>

<!-- jquery -->
<script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://maxcdn.bootstrapcdn.com/bootswatch/3.3.7/yeti/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">

<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script>

<!-- sticky kit -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



<meta property="og:title" content="Function reference" />



<!-- mathjax -->
<script src='https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML'></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-index">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">celda</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/celda-analysis.html">Analyzing single-cell RNA-seq count data with Celda</a>
    </li>
  </ul>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="contents col-md-9">
    <div class="page-header">
      <h1>Reference</h1>
    </div>

    <table class="ref-index">

    <colgroup>
      
      <col class="alias" />
      <col class="title" />
    </colgroup>

    <tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-function-documentation" class="hasAnchor"><a href="#section-function-documentation" class="anchor"></a>Function Documentation</h2>
          <p class="section-desc"><p>Man pages for all functions exported by celda</p></p>
        </th>
      </tr>
      <tr>
        
        <td>
          <p><code><a href="available_models.html">available_models</a></code> </p>
        </td>
        <td><p>available models</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="calc.perplexity.html">calc.perplexity</a></code> </p>
        </td>
        <td><p>calc.perplexity</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="calc.perplexity2.html">calc.perplexity2</a></code> </p>
        </td>
        <td><p>calc.perplexity2</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="calculateLoglikFromVariables.celda_C.html">calculateLoglikFromVariables.celda_C()</a></code> </p>
        </td>
        <td><p>Calculate the celda_C log likelihood for user-provided cluster assignments</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="calculateLoglikFromVariables.celda_CG.html">calculateLoglikFromVariables.celda_CG()</a></code> </p>
        </td>
        <td><p>Calculate log lileklihood for the celda Cell and Gene clustering model, given a set of cell / gene cluster assignments</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="calculateLoglikFromVariables.celda_G.html">calculateLoglikFromVariables.celda_G()</a></code> </p>
        </td>
        <td><p>Calculate the celda_G log likelihood for user-provided cluster assignments</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="calculateLoglikFromVariables.html">calculateLoglikFromVariables()</a></code> </p>
        </td>
        <td><p>Calculate a log-likelihood for a user-provided cluster assignment and count matrix, per the desired celda model.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="calculatePerplexity.celda_C.html">calculatePerplexity(<i>&lt;celda_C&gt;</i>)</a></code> </p>
        </td>
        <td><p>Calculate the perplexity from a single celda model</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="calculatePerplexity.celda_CG.html">calculatePerplexity(<i>&lt;celda_CG&gt;</i>)</a></code> </p>
        </td>
        <td><p>Calculate the perplexity from a single celda model</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="calculatePerplexity.celda_G.html">calculatePerplexity(<i>&lt;celda_G&gt;</i>)</a></code> </p>
        </td>
        <td><p>Calculate the perplexity from a single celda model</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="calculatePerplexity.html">calculatePerplexity()</a></code> </p>
        </td>
        <td><p>Calculate the perplexity from a single celda model</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="calculatePerplexityWithResampling.html">calculatePerplexityWithResampling()</a></code> </p>
        </td>
        <td><p>Calculate and visualize perplexity of all models in a celda_list, with count resampling</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="calculateTsne.html">calculateTsne()</a></code> </p>
        </td>
        <td><p>Run the t-SNE algorithm for dimensionality reduction</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="cC.decomposeCounts.html">cC.decomposeCounts()</a></code> </p>
        </td>
        <td><p>Takes raw counts matrix and converts it to a series of matrices needed for log likelihood calculation</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="cCG.decomposeCounts.html">cCG.decomposeCounts()</a></code> </p>
        </td>
        <td><p>Takes raw counts matrix and converts it to a series of matrices needed for log likelihood calculation</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="celda.html">celda()</a></code> </p>
        </td>
        <td><p>Deprecation warning for old grid search function</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="celdaGridSearch.html">celdaGridSearch()</a></code> </p>
        </td>
        <td><p>Run the celda Bayesian hierarchical model on a matrix of counts.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="celdaHeatmap.celda_C.html">celdaHeatmap(<i>&lt;celda_C&gt;</i>)</a></code> </p>
        </td>
        <td><p>celdaHeatmap for celda Cell clustering function</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="celdaHeatmap.celda_CG.html">celdaHeatmap(<i>&lt;celda_CG&gt;</i>)</a></code> </p>
        </td>
        <td><p>celdaHeatmap for celda Cell and Gene clustering model.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="celdaHeatmap.celda_G.html">celdaHeatmap(<i>&lt;celda_G&gt;</i>)</a></code> </p>
        </td>
        <td><p>celdaHeatmap for celda Gene clustering model</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="celdaHeatmap.html">celdaHeatmap()</a></code> </p>
        </td>
        <td><p>Render a stylable heatmap of count data based on celda clustering results.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="celdaProbabilityMap.celda_C.html">celdaProbabilityMap(<i>&lt;celda_C&gt;</i>)</a></code> </p>
        </td>
        <td><p>Renders probability and relative expression heatmaps to visualize the relationship between feature modules and cell populations.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="celdaProbabilityMap.celda_CG.html">celdaProbabilityMap(<i>&lt;celda_CG&gt;</i>)</a></code> </p>
        </td>
        <td><p>Renders probability and relative expression heatmaps to visualize the relationship between feature modules and cell populations.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="celdaProbabilityMap.html">celdaProbabilityMap()</a></code> </p>
        </td>
        <td><p>Renders probability and relative expression heatmaps to visualize the relationship between feature modules and cell populations.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="celdaTsne.celda_C.html">celdaTsne(<i>&lt;celda_C&gt;</i>)</a></code> </p>
        </td>
        <td><p>Embeds cells in two dimensions using tSNE based on celda_C results.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="celdaTsne.celda_CG.html">celdaTsne(<i>&lt;celda_CG&gt;</i>)</a></code> </p>
        </td>
        <td><p>Embeds cells in two dimensions using tSNE based on celda_CG results.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="celdaTsne.celda_G.html">celdaTsne(<i>&lt;celda_G&gt;</i>)</a></code> </p>
        </td>
        <td><p>Embeds cells in two dimensions using tSNE based on celda_CG results.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="celdaTsne.html">celdaTsne()</a></code> </p>
        </td>
        <td><p>Embeds cells in two dimensions using tSNE based on celda_CG results.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="celda_C.html">celda_C()</a></code> </p>
        </td>
        <td><p>celda Cell Clustering Model</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="celda_CG.html">celda_CG()</a></code> </p>
        </td>
        <td><p>celda Cell and Gene Clustering Model</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="celda_G.html">celda_G()</a></code> </p>
        </td>
        <td><p>celda Gene Clustering Model</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="cG.decomposeCounts.html">cG.decomposeCounts()</a></code> </p>
        </td>
        <td><p>Takes raw counts matrix and converts it to a series of matrices needed for log likelihood calculation</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="clusterProbability.celda_C.html">clusterProbability(<i>&lt;celda_C&gt;</i>)</a></code> </p>
        </td>
        <td><p>Calculates the conditional probability of each cell belong to each cluster given all other cluster assignments</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="clusterProbability.celda_CG.html">clusterProbability(<i>&lt;celda_CG&gt;</i>)</a></code> </p>
        </td>
        <td><p>Calculates the conditional probability of each cell belong to each cluster given all other cluster assignments</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="clusterProbability.celda_G.html">clusterProbability(<i>&lt;celda_G&gt;</i>)</a></code> </p>
        </td>
        <td><p>Calculates the conditional probability of each cell belong to each cluster given all other cluster assignments</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="clusterProbability.html">clusterProbability()</a></code> </p>
        </td>
        <td><p>Get the probability of the cluster assignments generated during a celda run.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="compareCountMatrix.html">compareCountMatrix()</a></code> </p>
        </td>
        <td><p>Check whether a count matrix was the one used in a given celda run</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="completeLogLikelihood.html">completeLogLikelihood()</a></code> </p>
        </td>
        <td><p>Get the complete log likelihood for a given celda model.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="differentialExpression.html">differentialExpression()</a></code> </p>
        </td>
        <td><p>Gene expression markers for cell clusters using MAST</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="distinct_colors.html">distinct_colors()</a></code> </p>
        </td>
        <td><p>Generate a distinct palette for coloring different clusters</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="eigenMatMultInt.html">eigenMatMultInt()</a></code> </p>
        </td>
        <td><p>Fast matrix multiplication for double x int</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="factorizeMatrix.celda_C.html">factorizeMatrix(<i>&lt;celda_C&gt;</i>)</a></code> </p>
        </td>
        <td><p>Generate factorized matrices showing each feature's influence on the celda_C model clustering</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="factorizeMatrix.celda_CG.html">factorizeMatrix(<i>&lt;celda_CG&gt;</i>)</a></code> </p>
        </td>
        <td><p>Generate factorized matrices showing each feature's influence on the celda_CG model clustering</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="factorizeMatrix.celda_G.html">factorizeMatrix(<i>&lt;celda_G&gt;</i>)</a></code> </p>
        </td>
        <td><p>Generate factorized matrices showing each feature's influence on the celda_G model clustering</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="factorizeMatrix.html">factorizeMatrix()</a></code> </p>
        </td>
        <td><p>Generate factorized matrices showing each feature's influence on cell / gene clustering</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="fastNormProp.html">fastNormProp()</a></code> </p>
        </td>
        <td><p>Fast normalization for numeric matrix</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="fastNormPropLog.html">fastNormPropLog()</a></code> </p>
        </td>
        <td><p>Fast normalization for numeric matrix</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="fastNormPropSqrt.html">fastNormPropSqrt()</a></code> </p>
        </td>
        <td><p>Fast normalization for numeric matrix</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="featureModuleLookup.celda_C.html">featureModuleLookup(<i>&lt;celda_C&gt;</i>)</a></code> </p>
        </td>
        <td><p>Obtain the gene module of a gene of interest</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="featureModuleLookup.celda_CG.html">featureModuleLookup(<i>&lt;celda_CG&gt;</i>)</a></code> </p>
        </td>
        <td><p>Obtain the feature module of a feature of interest</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="featureModuleLookup.celda_G.html">featureModuleLookup(<i>&lt;celda_G&gt;</i>)</a></code> </p>
        </td>
        <td><p>Obtain the gene module of a gene of interest</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="featureModuleLookup.html">featureModuleLookup()</a></code> </p>
        </td>
        <td><p>Obtain the gene module of a gene of interest</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="finalLogLikelihood.html">finalLogLikelihood()</a></code> </p>
        </td>
        <td><p>Get the log likelihood from the final iteration of Gibbs sampling
for a given celda model.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="moduleHeatmap.html">moduleHeatmap()</a></code> </p>
        </td>
        <td><p>Transcriptional state heatmap</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="normalizeCounts.html">normalizeCounts()</a></code> </p>
        </td>
        <td><p>Performs normalization, transformation, and/or scaling on a counts matrix</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="pbmc_res.html">pbmc_res</a></code> </p>
        </td>
        <td><p>pbmc_res</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="pbmc_select.html">pbmc_select</a></code> </p>
        </td>
        <td><p>pbmc_select</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plotDimReduceCluster.html">plotDimReduceCluster()</a></code> </p>
        </td>
        <td><p>Create a scatterplot based on celda cluster labels.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plotDimReduceGene.html">plotDimReduceGene()</a></code> </p>
        </td>
        <td><p>Create a scatterplot for each row of a normalized gene expression matrix where x and y axis are from a data dimensionality reduction tool.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plotDimReduceGrid.html">plotDimReduceGrid()</a></code> </p>
        </td>
        <td><p>Create a scatterplot given two dimensions from a data dimensionality reduction tool (e.g tSNE)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plotDimReduceState.html">plotDimReduceState()</a></code> </p>
        </td>
        <td><p>Create a scatterplot based off of a matrix containing the celda state probabilities per cell.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plotGridSearchPerplexity.celda_C.html">plotGridSearchPerplexity(<i>&lt;celda_C&gt;</i>)</a></code> </p>
        </td>
        <td><p>Plot perplexity as a function of K from celda_C models</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plotGridSearchPerplexity.celda_CG.html">plotGridSearchPerplexity(<i>&lt;celda_CG&gt;</i>)</a></code> </p>
        </td>
        <td><p>Plot perplexity as a function of K and L from celda_CG models</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plotGridSearchPerplexity.celda_G.html">plotGridSearchPerplexity(<i>&lt;celda_G&gt;</i>)</a></code> </p>
        </td>
        <td><p>Plot perplexity as a function of L from a celda_G model</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plotGridSearchPerplexity.html">plotGridSearchPerplexity()</a></code> </p>
        </td>
        <td><p>Visualize perplexity of every model in a celda_list, by unique K/L combinations</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plotHeatmap.html">plotHeatmap()</a></code> </p>
        </td>
        <td><p>Renders a heatmap based on a matrix of counts where rows are features and columns are cells.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="recodeClusterY.html">recodeClusterY()</a></code> </p>
        </td>
        <td><p>Re-code gene cluster labels by provided mapping scheme</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="recodeClusterZ.html">recodeClusterZ()</a></code> </p>
        </td>
        <td><p>Re-code cell cluster labels by provided mapping scheme</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="runParams.html">runParams()</a></code> </p>
        </td>
        <td><p>Get run parameters for a celda run.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="sample.cells.html">sample.cells</a></code> </p>
        </td>
        <td><p>sample.cells</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="selectBestModel.html">selectBestModel()</a></code> </p>
        </td>
        <td><p>Select models with the best log likelihood from a 'celda_list' object gererated by celdaGridSearch.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="semi_pheatmap.html">semi_pheatmap()</a></code> </p>
        </td>
        <td><p>A function to draw clustered heatmaps.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="simulateCells.celda_C.html">simulateCells(<i>&lt;celda_C&gt;</i>)</a></code> </p>
        </td>
        <td><p>Simulate cells from the cell clustering generative model</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="simulateCells.celda_CG.html">simulateCells(<i>&lt;celda_CG&gt;</i>)</a></code> </p>
        </td>
        <td><p>Simulate cells from the cell/feature bi-clustering generative model</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="simulateCells.celda_G.html">simulateCells(<i>&lt;celda_G&gt;</i>)</a></code> </p>
        </td>
        <td><p>Simulate cells from the feature clustering generative model</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="simulateCells.html">simulateCells()</a></code> </p>
        </td>
        <td><p>Simulate count data from the celda generative models.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="subsetCeldaList.html">subsetCeldaList()</a></code> </p>
        </td>
        <td><p>Select a subset of models from a 'celda_list' object generated by celdaGridSearch.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="topRank.html">topRank()</a></code> </p>
        </td>
        <td><p>Identify features with the highest influence on clustering.</p></td>
      </tr>
    </tbody>
    </table>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#section-function-documentation">Function Documentation</a></li>
    </ul>
  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Joshua Campbell, Sean Corbett, Yusuke Koga.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
   </div>

  

  </body>
</html>

