<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html>
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Renders a heatmap based on a matrix of counts where rows are features and columns are cells. — plotHeatmap • celda</title>

<!-- jquery -->
<script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://maxcdn.bootstrapcdn.com/bootswatch/3.3.7/yeti/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">

<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script>

<!-- sticky kit -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



<meta property="og:title" content="Renders a heatmap based on a matrix of counts where rows are features and columns are cells. — plotHeatmap" />

<meta property="og:description" content="Renders a heatmap based on a matrix of counts where rows are features and columns are cells." />
<meta name="twitter:card" content="summary" />



<!-- mathjax -->
<script src='https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML'></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">celda</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/celda-analysis.html">Analyzing single-cell RNA-seq count data with Celda</a>
    </li>
  </ul>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Renders a heatmap based on a matrix of counts where rows are features and columns are cells.</h1>
    
    <div class="hidden name"><code>plotHeatmap.Rd</code></div>
    </div>

    <div class="ref-description">
    
    <p>Renders a heatmap based on a matrix of counts where rows are features and columns are cells.</p>
    
    </div>

    <pre class="usage"><span class='fu'>plotHeatmap</span>(<span class='no'>counts</span>, <span class='kw'>z</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>y</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>scale.row</span> <span class='kw'>=</span> <span class='no'>scale</span>, <span class='kw'>trim</span> <span class='kw'>=</span> <span class='fu'>c</span>(-<span class='fl'>2</span>,
  <span class='fl'>2</span>), <span class='kw'>feature.ix</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>cell.ix</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>cluster.feature</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
  <span class='kw'>cluster.cell</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>color.scheme</span> <span class='kw'>=</span> <span class='fu'>c</span>(<span class='st'>"divergent"</span>, <span class='st'>"sequential"</span>),
  <span class='kw'>color.scheme.symmetric</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>color.scheme.center</span> <span class='kw'>=</span> <span class='fl'>0</span>, <span class='kw'>col</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>annotation.cell</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>annotation.feature</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>annotation.color</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>breaks</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>legend</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
  <span class='kw'>annotation.legend</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>annotation.names.feature</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
  <span class='kw'>annotation.names.cell</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>show.names.feature</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>show.names.cell</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>, <span class='kw'>hclust.method</span> <span class='kw'>=</span> <span class='st'>"ward.D2"</span>,
  <span class='kw'>treeheight.feature</span> <span class='kw'>=</span> <span class='fu'>ifelse</span>(<span class='no'>cluster.feature</span>, <span class='fl'>50</span>, <span class='fl'>0</span>),
  <span class='kw'>treeheight.cell</span> <span class='kw'>=</span> <span class='fu'>ifelse</span>(<span class='no'>cluster.cell</span>, <span class='fl'>50</span>, <span class='fl'>0</span>), <span class='kw'>silent</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>, <span class='no'>...</span>)</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>counts</th>
      <td><p>Numeric matrix. Normalized counts matrix where rows represent features and columns represent cells. .</p></td>
    </tr>
    <tr>
      <th>z</th>
      <td><p>Numeric vector. Denotes cell population labels.</p></td>
    </tr>
    <tr>
      <th>y</th>
      <td><p>Numeric vector. Denotes feature module labels.</p></td>
    </tr>
    <tr>
      <th>scale.row</th>
      <td><p>Function; A function to scale each individual row. Set to NULL to disable. Occurs after normalization and log transformation. Defualt is 'scale' and thus will Z-score transform each row.</p></td>
    </tr>
    <tr>
      <th>trim</th>
      <td><p>Numeric vector. Vector of length two that specifies the lower and upper bounds for the data. This threshold is applied after row scaling. Set to NULL to disable. Default c(-2,2).</p></td>
    </tr>
    <tr>
      <th>feature.ix</th>
      <td><p>Integer vector. Select features for display in heatmap. If NULL, no subsetting will be performed. Default NULL.</p></td>
    </tr>
    <tr>
      <th>cell.ix</th>
      <td><p>Integer vector. Select cells for display in heatmap. If NULL, no subsetting will be performed. Default NULL.</p></td>
    </tr>
    <tr>
      <th>cluster.feature</th>
      <td><p>Logical. Determines whether rows should be clustered. Default TRUE.</p></td>
    </tr>
    <tr>
      <th>cluster.cell</th>
      <td><p>Logical. Determines whether columns should be clustered. Default TRUE.</p></td>
    </tr>
    <tr>
      <th>color.scheme</th>
      <td><p>"Character. One of ""divergent"" or ""sequential"". A ""divergent"" scheme is best for highlighting relative data (denoted by 'color.scheme.center') such as gene expression data that has been normalized and centered. A ""sequential"" scheme is best for highlighting data that are ordered low to high such as raw counts or probabilities. Default "divergent".</p></td>
    </tr>
    <tr>
      <th>color.scheme.symmetric</th>
      <td><p>Logical. When the color.scheme is "divergent" and the data contains both positive and negative numbers, TRUE indicates that the color scheme should be symmetric from [-max(abs(data)),max(abs(data))]. For example, if the data ranges goes from -1.5 to 2, then setting this to TRUE will force the color scheme to range from -2 to 2. Default TRUE.</p></td>
    </tr>
    <tr>
      <th>color.scheme.center</th>
      <td><p>Numeric. Indicates the center of a "divergent" color.scheme. Default 0.</p></td>
    </tr>
    <tr>
      <th>col</th>
      <td><p>Color for the heatmap.</p></td>
    </tr>
    <tr>
      <th>annotation.cell</th>
      <td><p>Data frame. Additional annotations for each cell will be shown in the column color bars. The format of the data frame should be one row for each cell and one column for each annotation. Numeric variables will be displayed as continuous color bars and factors will be displayed as discrete color bars. Default NULL.</p></td>
    </tr>
    <tr>
      <th>annotation.feature</th>
      <td><p>A data frame for the feature annotations (rows).</p></td>
    </tr>
    <tr>
      <th>annotation.color</th>
      <td><p>List. Contains color scheme for all annotations. See `?pheatmap` for more details.</p></td>
    </tr>
    <tr>
      <th>breaks</th>
      <td><p>Numeric vector. A sequence of numbers that covers the range of values in the normalized `counts`. Values in the normalized `matrix` are assigned to each bin in `breaks`. Each break is assigned to a unique color from `col`. If NULL, then breaks are calculated automatically. Default NULL.</p></td>
    </tr>
    <tr>
      <th>legend</th>
      <td><p>Logical. Determines whether legend should be drawn. Default TRUE.</p></td>
    </tr>
    <tr>
      <th>annotation.legend</th>
      <td><p>Logical. Whether legend for all annotations should be drawn. Default TRUE.</p></td>
    </tr>
    <tr>
      <th>annotation.names.feature</th>
      <td><p>Logical. Whether the names for features should be shown. Default TRUE.</p></td>
    </tr>
    <tr>
      <th>annotation.names.cell</th>
      <td><p>Logical. Whether the names for cells should be shown. Default TRUE.</p></td>
    </tr>
    <tr>
      <th>show.names.feature</th>
      <td><p>Logical. Specifies if feature names should be shown. Default TRUE.</p></td>
    </tr>
    <tr>
      <th>show.names.cell</th>
      <td><p>Logical. Specifies if cell names should be shown. Default FALSE.</p></td>
    </tr>
    <tr>
      <th>hclust.method</th>
      <td><p>Character. Specifies the method to use for the 'hclust' function. See `?hclust` for possible values. Default "ward.D2".</p></td>
    </tr>
    <tr>
      <th>treeheight.feature</th>
      <td><p>Numeric. Width of the feature dendrogram. Set to 0 to disable plotting of this dendrogram. Default: if cluster.feature == TRUE, then treeheight.feature = 50, else treeheight.feature = 0.</p></td>
    </tr>
    <tr>
      <th>treeheight.cell</th>
      <td><p>Numeric. Height of the cell dendrogram. Set to 0 to disable plotting of this dendrogram. Default: if cluster.cell == TRUE, then treeheight.cell = 50, else treeheight.cell = 0.</p></td>
    </tr>
    <tr>
      <th>silent</th>
      <td><p>Logical. Whether to plot the heatmap.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>Other arguments to be passed to underlying pheatmap function.</p></td>
    </tr>
    </table>
    

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='no'>celda.sim</span> <span class='kw'>=</span> <span class='fu'><a href='simulateCells.html'>simulateCells</a></span>(<span class='kw'>model</span><span class='kw'>=</span><span class='st'>"celda_CG"</span>)
<span class='no'>celda.mod</span> <span class='kw'>=</span> <span class='fu'><a href='celda_CG.html'>celda_CG</a></span>(<span class='no'>celda.sim</span>$<span class='no'>counts</span>, <span class='kw'>K</span><span class='kw'>=</span><span class='no'>celda.sim</span>$<span class='no'>K</span>, <span class='kw'>L</span><span class='kw'>=</span><span class='no'>celda.sim</span>$<span class='no'>L</span>,
                     <span class='kw'>sample.label</span><span class='kw'>=</span><span class='no'>celda.sim</span>$<span class='no'>sample.label</span>,
                     <span class='kw'>max.iter</span><span class='kw'>=</span><span class='fl'>2</span>, <span class='kw'>nchains</span><span class='kw'>=</span><span class='fl'>1</span>)</div><div class='output co'>#&gt; <span class='message'>--------------------------------------------------------------------</span></div><div class='output co'>#&gt; <span class='message'>Starting Celda_CG: Clustering cells and genes.</span></div><div class='output co'>#&gt; <span class='message'>--------------------------------------------------------------------</span></div><div class='output co'>#&gt; <span class='message'>Thu Sep 06 12:59:25 2018 .. Initializing chain 1 with 'random' (seed=12345)</span></div><div class='output co'>#&gt; <span class='message'>Thu Sep 06 12:59:25 2018 .... Completed iteration: 1 | logLik: -1302025.22571701</span></div><div class='output co'>#&gt; <span class='message'>Thu Sep 06 12:59:25 2018 .... Determining if any cell clusters should be split.</span></div><div class='output co'>#&gt; <span class='message'>Thu Sep 06 12:59:25 2018 .... Cluster 2 was reassigned and cluster 3 was split in two.</span></div><div class='output co'>#&gt; <span class='message'>Thu Sep 06 12:59:25 2018 .... Determining if any gene clusters should be split.</span></div><div class='output co'>#&gt; <span class='message'>Thu Sep 06 12:59:26 2018 .... Cluster 8 was reassigned and cluster 7 was split in two.</span></div><div class='output co'>#&gt; <span class='message'>Thu Sep 06 12:59:26 2018 .... Completed iteration: 2 | logLik: -1270140.00618477</span></div><div class='output co'>#&gt; <span class='message'>Thu Sep 06 12:59:26 2018 .. Finished chain 1 with seed 12345</span></div><div class='output co'>#&gt; <span class='message'>--------------------------------------------------------------------</span></div><div class='output co'>#&gt; <span class='message'>Completed Celda_CG. Total time: 1.014419 secs</span></div><div class='output co'>#&gt; <span class='message'>--------------------------------------------------------------------</span></div><div class='input'><span class='fu'>plotHeatmap</span>(<span class='no'>celda.sim</span>$<span class='no'>counts</span>, <span class='kw'>z</span><span class='kw'>=</span><span class='no'>celda.mod</span>$<span class='no'>z</span>, <span class='kw'>y</span><span class='kw'>=</span><span class='no'>celda.mod</span>$<span class='no'>y</span>)</div><div class='img'><img src='plotHeatmap-1.png' alt='' width='700' height='433' /></div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
            
      <li><a href="#examples">Examples</a></li>
    </ul>

  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Joshua Campbell, Sean Corbett, Yusuke Koga.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
   </div>

  

  </body>
</html>

