<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html>
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>A function to draw clustered heatmaps. — semi_pheatmap • celda</title>

<!-- jquery -->
<script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://maxcdn.bootstrapcdn.com/bootswatch/3.3.7/yeti/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">

<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script>

<!-- sticky kit -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



<meta property="og:title" content="A function to draw clustered heatmaps. — semi_pheatmap" />

<meta property="og:description" content="A function to draw clustered heatmaps where one has better control over some graphical 
parameters such as cell size, etc." />
<meta name="twitter:card" content="summary" />



<!-- mathjax -->
<script src='https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML'></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">celda</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/celda-analysis.html">Analyzing single-cell RNA-seq count data with Celda</a>
    </li>
  </ul>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>A function to draw clustered heatmaps.</h1>
    
    <div class="hidden name"><code>semi_pheatmap.Rd</code></div>
    </div>

    <div class="ref-description">
    
    <p>A function to draw clustered heatmaps where one has better control over some graphical 
parameters such as cell size, etc.</p>
    
    </div>

    <pre class="usage"><span class='fu'>semi_pheatmap</span>(<span class='no'>mat</span>, <span class='kw'>color</span> <span class='kw'>=</span> <span class='fu'>colorRampPalette</span>(<span class='fu'>rev</span>(<span class='fu'>brewer.pal</span>(<span class='kw'>n</span> <span class='kw'>=</span> <span class='fl'>7</span>, <span class='kw'>name</span> <span class='kw'>=</span>
  <span class='st'>"RdYlBu"</span>)))(<span class='fl'>100</span>), <span class='kw'>kmeans_k</span> <span class='kw'>=</span> <span class='fl'>NA</span>, <span class='kw'>breaks</span> <span class='kw'>=</span> <span class='fl'>NA</span>, <span class='kw'>border_color</span> <span class='kw'>=</span> <span class='st'>"grey60"</span>,
  <span class='kw'>cellwidth</span> <span class='kw'>=</span> <span class='fl'>NA</span>, <span class='kw'>cellheight</span> <span class='kw'>=</span> <span class='fl'>NA</span>, <span class='kw'>scale</span> <span class='kw'>=</span> <span class='st'>"none"</span>, <span class='kw'>cluster_rows</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
  <span class='kw'>cluster_cols</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>clustering_distance_rows</span> <span class='kw'>=</span> <span class='st'>"euclidean"</span>,
  <span class='kw'>clustering_distance_cols</span> <span class='kw'>=</span> <span class='st'>"euclidean"</span>, <span class='kw'>clustering_method</span> <span class='kw'>=</span> <span class='st'>"complete"</span>,
  <span class='kw'>clustering_callback</span> <span class='kw'>=</span> <span class='no'>identity2</span>, <span class='kw'>cutree_rows</span> <span class='kw'>=</span> <span class='fl'>NA</span>, <span class='kw'>cutree_cols</span> <span class='kw'>=</span> <span class='fl'>NA</span>,
  <span class='kw'>treeheight_row</span> <span class='kw'>=</span> <span class='fu'>ifelse</span>(<span class='no'>cluster_rows</span>, <span class='fl'>50</span>, <span class='fl'>0</span>),
  <span class='kw'>treeheight_col</span> <span class='kw'>=</span> <span class='fu'>ifelse</span>(<span class='no'>cluster_cols</span>, <span class='fl'>50</span>, <span class='fl'>0</span>), <span class='kw'>legend</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
  <span class='kw'>legend_breaks</span> <span class='kw'>=</span> <span class='fl'>NA</span>, <span class='kw'>legend_labels</span> <span class='kw'>=</span> <span class='fl'>NA</span>, <span class='kw'>annotation_row</span> <span class='kw'>=</span> <span class='fl'>NA</span>,
  <span class='kw'>annotation_col</span> <span class='kw'>=</span> <span class='fl'>NA</span>, <span class='kw'>annotation</span> <span class='kw'>=</span> <span class='fl'>NA</span>, <span class='kw'>annotation_colors</span> <span class='kw'>=</span> <span class='fl'>NA</span>,
  <span class='kw'>annotation_legend</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>annotation_names_row</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
  <span class='kw'>annotation_names_col</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>drop_levels</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>show_rownames</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
  <span class='kw'>show_colnames</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>main</span> <span class='kw'>=</span> <span class='fl'>NA</span>, <span class='kw'>fontsize</span> <span class='kw'>=</span> <span class='fl'>10</span>,
  <span class='kw'>fontsize_row</span> <span class='kw'>=</span> <span class='no'>fontsize</span>, <span class='kw'>fontsize_col</span> <span class='kw'>=</span> <span class='no'>fontsize</span>,
  <span class='kw'>display_numbers</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>, <span class='kw'>number_format</span> <span class='kw'>=</span> <span class='st'>"%.2f"</span>,
  <span class='kw'>number_color</span> <span class='kw'>=</span> <span class='st'>"grey30"</span>, <span class='kw'>fontsize_number</span> <span class='kw'>=</span> <span class='fl'>0.8</span> * <span class='no'>fontsize</span>,
  <span class='kw'>gaps_row</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>gaps_col</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>labels_row</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>labels_col</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>filename</span> <span class='kw'>=</span> <span class='fl'>NA</span>, <span class='kw'>width</span> <span class='kw'>=</span> <span class='fl'>NA</span>, <span class='kw'>height</span> <span class='kw'>=</span> <span class='fl'>NA</span>,
  <span class='kw'>silent</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>, <span class='no'>row_label</span>, <span class='no'>col_label</span>, <span class='no'>...</span>)</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>mat</th>
      <td><p>numeric matrix of the values to be plotted.</p></td>
    </tr>
    <tr>
      <th>color</th>
      <td><p>vector of colors used in heatmap.</p></td>
    </tr>
    <tr>
      <th>kmeans_k</th>
      <td><p>the number of kmeans clusters to make, if we want to agggregate the 
rows before drawing heatmap. If NA then the rows are not aggregated.</p></td>
    </tr>
    <tr>
      <th>breaks</th>
      <td><p>Numeric vector. A sequence of numbers that covers the range of values in the normalized `counts`. Values in the normalized `matrix` are assigned to each bin in `breaks`. Each break is assigned to a unique color from `col`. If NULL, then breaks are calculated automatically. Default NULL.</p></td>
    </tr>
    <tr>
      <th>border_color</th>
      <td><p>color of cell borders on heatmap, use NA if no border should be 
drawn.</p></td>
    </tr>
    <tr>
      <th>cellwidth</th>
      <td><p>individual cell width in points. If left as NA, then the values 
depend on the size of plotting window.</p></td>
    </tr>
    <tr>
      <th>cellheight</th>
      <td><p>individual cell height in points. If left as NA, 
then the values depend on the size of plotting window.</p></td>
    </tr>
    <tr>
      <th>scale</th>
      <td><p>character indicating if the values should be centered and scaled in 
either the row direction or the column direction, or none. Corresponding values are 
<code>"row"</code>, <code>"column"</code> and <code>"none"</code></p></td>
    </tr>
    <tr>
      <th>cluster_rows</th>
      <td><p>boolean values determining if rows should be clustered or <code>hclust</code> object,</p></td>
    </tr>
    <tr>
      <th>cluster_cols</th>
      <td><p>boolean values determining if columns should be clustered or <code>hclust</code> object.</p></td>
    </tr>
    <tr>
      <th>clustering_distance_rows</th>
      <td><p>distance measure used in clustering rows. Possible 
values are <code>"correlation"</code> for Pearson correlation and all the distances 
supported by <code>dist</code>, such as <code>"euclidean"</code>, etc. If the value is none 
of the above it is assumed that a distance matrix is provided.</p></td>
    </tr>
    <tr>
      <th>clustering_distance_cols</th>
      <td><p>distance measure used in clustering columns. Possible 
values the same as for clustering_distance_rows.</p></td>
    </tr>
    <tr>
      <th>clustering_method</th>
      <td><p>clustering method used. Accepts the same values as 
<code>hclust</code>.</p></td>
    </tr>
    <tr>
      <th>clustering_callback</th>
      <td><p>callback function to modify the clustering. Is 
called with two parameters: original <code>hclust</code> object and the matrix 
used for clustering. Must return a <code>hclust</code> object.</p></td>
    </tr>
    <tr>
      <th>cutree_rows</th>
      <td><p>number of clusters the rows are divided into, based on the
 hierarchical clustering (using cutree), if rows are not clustered, the 
argument is ignored</p></td>
    </tr>
    <tr>
      <th>cutree_cols</th>
      <td><p>similar to <code>cutree_rows</code>, but for columns</p></td>
    </tr>
    <tr>
      <th>treeheight_row</th>
      <td><p>the height of a tree for rows, if these are clustered. 
Default value 50 points.</p></td>
    </tr>
    <tr>
      <th>treeheight_col</th>
      <td><p>the height of a tree for columns, if these are clustered. 
Default value 50 points.</p></td>
    </tr>
    <tr>
      <th>legend</th>
      <td><p>logical to determine if legend should be drawn or not.</p></td>
    </tr>
    <tr>
      <th>legend_breaks</th>
      <td><p>vector of breakpoints for the legend.</p></td>
    </tr>
    <tr>
      <th>legend_labels</th>
      <td><p>vector of labels for the <code>legend_breaks</code>.</p></td>
    </tr>
    <tr>
      <th>annotation_row</th>
      <td><p>data frame that specifies the annotations shown on left
 side of the heatmap. Each row defines the features for a specific row. The 
rows in the data and in the annotation are matched using corresponding row
 names. Note that color schemes takes into account if variable is continuous
 or discrete.</p></td>
    </tr>
    <tr>
      <th>annotation_col</th>
      <td><p>similar to annotation_row, but for columns.</p></td>
    </tr>
    <tr>
      <th>annotation</th>
      <td><p>deprecated parameter that currently sets the annotation_col if it is missing</p></td>
    </tr>
    <tr>
      <th>annotation_colors</th>
      <td><p>list for specifying annotation_row and 
annotation_col track colors manually. It is  possible to define the colors 
for only some of the features. Check examples for  details.</p></td>
    </tr>
    <tr>
      <th>annotation_legend</th>
      <td><p>boolean value showing if the legend for annotation 
tracks should be drawn.</p></td>
    </tr>
    <tr>
      <th>annotation_names_row</th>
      <td><p>boolean value showing if the names for row annotation 
tracks should be drawn.</p></td>
    </tr>
    <tr>
      <th>annotation_names_col</th>
      <td><p>boolean value showing if the names for column annotation 
tracks should be drawn.</p></td>
    </tr>
    <tr>
      <th>drop_levels</th>
      <td><p>logical to determine if unused levels are also shown in 
the legend</p></td>
    </tr>
    <tr>
      <th>show_rownames</th>
      <td><p>boolean specifying if column names are be shown.</p></td>
    </tr>
    <tr>
      <th>show_colnames</th>
      <td><p>boolean specifying if column names are be shown.</p></td>
    </tr>
    <tr>
      <th>main</th>
      <td><p>the title of the plot</p></td>
    </tr>
    <tr>
      <th>fontsize</th>
      <td><p>base fontsize for the plot</p></td>
    </tr>
    <tr>
      <th>fontsize_row</th>
      <td><p>fontsize for rownames (Default: fontsize)</p></td>
    </tr>
    <tr>
      <th>fontsize_col</th>
      <td><p>fontsize for colnames (Default: fontsize)</p></td>
    </tr>
    <tr>
      <th>display_numbers</th>
      <td><p>logical determining if the numeric values are also printed to 
the cells. If this is a matrix (with same dimensions as original matrix), the contents
of the matrix are shown instead of original values.</p></td>
    </tr>
    <tr>
      <th>number_format</th>
      <td><p>format strings (C printf style) of the numbers shown in cells. 
For example "<code>%.2f</code>" shows 2 decimal places and "<code>%.1e</code>" shows exponential 
notation (see more in <code>sprintf</code>).</p></td>
    </tr>
    <tr>
      <th>number_color</th>
      <td><p>color of the text</p></td>
    </tr>
    <tr>
      <th>fontsize_number</th>
      <td><p>fontsize of the numbers displayed in cells</p></td>
    </tr>
    <tr>
      <th>gaps_row</th>
      <td><p>vector of row indices that show shere to put gaps into
heatmap. Used only if the rows are not clustered. See <code>cutree_row</code>
to see how to introduce gaps to clustered rows.</p></td>
    </tr>
    <tr>
      <th>gaps_col</th>
      <td><p>similar to gaps_row, but for columns.</p></td>
    </tr>
    <tr>
      <th>labels_row</th>
      <td><p>custom labels for rows that are used instead of rownames.</p></td>
    </tr>
    <tr>
      <th>labels_col</th>
      <td><p>similar to labels_row, but for columns.</p></td>
    </tr>
    <tr>
      <th>filename</th>
      <td><p>file path where to save the picture. Filetype is decided by 
the extension in the path. Currently following formats are supported: png, pdf, tiff,
 bmp, jpeg. Even if the plot does not fit into the plotting window, the file size is 
calculated so that the plot would fit there, unless specified otherwise.</p></td>
    </tr>
    <tr>
      <th>width</th>
      <td><p>manual option for determining the output file width in inches.</p></td>
    </tr>
    <tr>
      <th>height</th>
      <td><p>manual option for determining the output file height in inches.</p></td>
    </tr>
    <tr>
      <th>silent</th>
      <td><p>do not draw the plot (useful when using the gtable output)</p></td>
    </tr>
    <tr>
      <th>row_label</th>
      <td><p>row cluster labels for semi-clustering</p></td>
    </tr>
    <tr>
      <th>col_label</th>
      <td><p>column cluster labels for semi-clustering</p></td>
    </tr>
    <tr>
      <th>&#8230;</th>
      <td><p>graphical parameters for the text used in plot. Parameters passed to 
<code>grid.text</code>, see <code>gpar</code>.</p></td>
    </tr>
    </table>
    
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>Invisibly a list of components</p><ul>
<li><p><code>tree_row</code> the clustering of rows as <code>hclust</code> object</p></li>
<li><p><code>tree_col</code> the clustering of columns as <code>hclust</code> object</p></li>
<li><p><code>kmeans</code> the kmeans clustering of rows if parameter <code>kmeans_k</code> was 
specified</p></li>
</ul>

    
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>The function also allows to aggregate the rows using kmeans clustering. This is 
advisable if number of rows is so big that R cannot handle their hierarchical 
clustering anymore, roughly more than 1000. Instead of showing all the rows 
separately one can cluster the rows in advance and show only the cluster centers. 
The number of clusters can be tuned with parameter kmeans_k.</p>
    

  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      
      <li><a href="#value">Value</a></li>

      <li><a href="#details">Details</a></li>
          </ul>

    <h2>Author</h2>
    
Raivo Kolde &lt;rkolde@gmail.com&gt;
#@examples
# Create test matrix
test = matrix(rnorm(200), 20, 10)
test[1:10, seq(1, 10, 2)] = test[1:10, seq(1, 10, 2)] + 3
test[11:20, seq(2, 10, 2)] = test[11:20, seq(2, 10, 2)] + 2
test[15:20, seq(2, 10, 2)] = test[15:20, seq(2, 10, 2)] + 4
colnames(test) = paste("Test", 1:10, sep = "")
rownames(test) = paste("Gene", 1:20, sep = "")

# Draw heatmaps
pheatmap(test)
pheatmap(test, kmeans_k = 2)
pheatmap(test, scale = "row", clustering_distance_rows = "correlation")
pheatmap(test, color = colorRampPalette(c("navy", "white", "firebrick3"))(50))
pheatmap(test, cluster_row = FALSE)
pheatmap(test, legend = FALSE)

# Show text within cells
pheatmap(test, display_numbers = TRUE)
pheatmap(test, display_numbers = TRUE, number_format = "%.1e")
pheatmap(test, display_numbers = matrix(ifelse(test &gt; 5, "*", ""), nrow(test)))
pheatmap(test, cluster_row = FALSE, legend_breaks = -1:4, legend_labels = c("0",
"1e-4", "1e-3", "1e-2", "1e-1", "1"))

# Fix cell sizes and save to file with correct size
pheatmap(test, cellwidth = 15, cellheight = 12, main = "Example heatmap")
pheatmap(test, cellwidth = 15, cellheight = 12, fontsize = 8, filename = "test.pdf")

# Generate annotations for rows and columns
annotation_col = data.frame(
                    CellType = factor(rep(c("CT1", "CT2"), 5)), 
                    Time = 1:5
                )
rownames(annotation_col) = paste("Test", 1:10, sep = "")

annotation_row = data.frame(
                    GeneClass = factor(rep(c("Path1", "Path2", "Path3"), c(10, 4, 6)))
                )
rownames(annotation_row) = paste("Gene", 1:20, sep = "")

# Display row and color annotations
pheatmap(test, annotation_col = annotation_col)
pheatmap(test, annotation_col = annotation_col, annotation_legend = FALSE)
pheatmap(test, annotation_col = annotation_col, annotation_row = annotation_row)


# Specify colors
ann_colors = list(
    Time = c("white", "firebrick"),
    CellType = c(CT1 = "#1B9E77", CT2 = "#D95F02"),
    GeneClass = c(Path1 = "#7570B3", Path2 = "#E7298A", Path3 = "#66A61E")
)

pheatmap(test, annotation_col = annotation_col, annotation_colors = ann_colors, main = "Title")
pheatmap(test, annotation_col = annotation_col, annotation_row = annotation_row, 
         annotation_colors = ann_colors)
pheatmap(test, annotation_col = annotation_col, annotation_colors = ann_colors[2]) 

# Gaps in heatmaps
pheatmap(test, annotation_col = annotation_col, cluster_rows = FALSE, gaps_row = c(10, 14))
pheatmap(test, annotation_col = annotation_col, cluster_rows = FALSE, gaps_row = c(10, 14), 
         cutree_col = 2)

# Show custom strings as row/col names
labels_row = c("", "", "", "", "", "", "", "", "", "", "", "", "", "", "", 
"", "", "Il10", "Il15", "Il1b")

pheatmap(test, annotation_col = annotation_col, labels_row = labels_row)

# Specifying clustering from distance matrix
drows = stats::dist(test, method = "minkowski")
dcols = stats::dist(t(test), method = "minkowski")
pheatmap(test, clustering_distance_rows = drows, clustering_distance_cols = dcols)

# Modify ordering of the clusters using clustering callback option
callback = function(hc, mat)
    sv = svd(t(mat))$v[,1]
    dend = reorder(as.dendrogram(hc), wts = sv)
    as.hclust(dend)


pheatmap(test, clustering_callback = callback)

dontrun
# Same using dendsort package
library(dendsort)

callback = function(hc, ...)dendsort(hc)
pheatmap(test, clustering_callback = callback)


  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Joshua Campbell, Sean Corbett, Yusuke Koga.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
   </div>

  

  </body>
</html>

